### ISSUE TITLE
Select All Matching does not paginate 

### ISSUE THREAD

**Issue by dillingham** [laravel/nova-issues#1097]
- Laravel Version: 5.7
- Nova Version: 1.2.0

### Description:

When you select all matching it should persist when clicking next
The checkboxes on the following pages should be checked also
Visually, the next page should be select all matching already
it should still be selected when you paginate
If you select all matching and want to inspect the next page it unchecks

### Steps To Reproduce:

 - Visit a resource with 25 + rows
 - Select All Matching
 - Click Next
 - See selection cleared

```php
// appending to pagination
// ?select_all_matching=true

$results->appends(request()->query())->links()
```

**Comment by waygou** [comment]
Data persistence across multi pagination would be great! Although complex to implement since data can change in parallel of the data selection ... meaning maybe a checkbox that was selected, no longer can be selected due to a data change.

**Comment by dillingham** [comment]
@waygou if something is select all matching, it’s all of the current filtered set. So I think it will always be consistent. 

Also, it’s more of a visual thing.. if select all matching is checked on previous page, mark it checked on this page..thus checking the checkboxes as expected.

**Comment by dkulyk** [comment]
@davidhemphill @dillingham proposed save "Select All Matching" between pages. What do you think about it?

**Comment by dillingham** [comment]
@davidhemphill It makes sense I think if you picture it from a "ok I selected all let me look at what I just selected, ok bottom of the page.. click next.. oh nooo its not selected anymore" perspective.

**Comment by jbrooksuk** [comment]
This issue relates to an unsupported version of Nova. If you're still experiencing this in the latest version of Nova, please open a new ticket, thanks!
